### PR TITLE
build/config & arch/xtensa:  Enable kernel tc for esp32 kit board.

### PR DESCRIPTION
**Pull request by xiarihf**
build/config & arch/xtensa:  Enable kernel tc for esp32 kit board.
  1. Change vdbg to svdbg in Xtensa code.
  2. Add Xtensa code branch in mm.h file.
  3. Add kernel tc configuartion for esp32 DevkitC & esp wrover kit board.

Signed-off-by: xiarihf qiang3.zhang@samsung.com

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a26b41f0c7d6d40ff325aa75aa81bb57f05a2f26] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5c2864476ef012b926d2fcf5ad495fff77d2d81f] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5c2864476ef012b926d2fcf5ad495fff77d2d81f] - Code Rule Check OK.

**Comment by sunghan-chang** [comment]
@xiarihf 
1. For compilation error on CI, I restarted it because it was timed-out.
2. Let's split svdbg changes and return address change as two different commits.
3. Let's use `tc` as a name of configs, not `kernel_tc`. We don't need to make a limitation for that. Later we can expand it to fs, nw and so on.

File: build/configs/esp32_DevKitC/kernel_tc/defconfig [hunk 1]
```diff
@@ -0,0 +1,839 @@
...
+CONFIG_ITC_KERN_PTHREAD_LOOP_SIZE=100
+# CONFIG_EXAMPLES_TESTCASE_NETWORK is not set
+# CONFIG_EXAMPLES_TESTCASE_TTRACE is not set
+# CONFIG_EXAMPLES_TESTCASE_ARASTORAGE_UTC is not set
+# CONFIG_EXAMPLES_TESTCASE_ARASTORAGE_ITC is not set
+# CONFIG_EXAMPLES_TESTCASE_AUDIO_UTC is not set
+# CONFIG_EXAMPLES_TESTCASE_AUDIO_ITC is not set
+# CONFIG_EXAMPLES_TESTCASE_SYSTEMIO_UTC is not set
+# CONFIG_EXAMPLES_TESTCASE_SYSTEMIO_ITC is not set
+# CONFIG_EXAMPLES_TESTCASE_WIFI_MANAGER_UTC is not set
```
Comment (xixidodo) [inline build/configs/esp32_DevKitC/kernel_tc/defconfig]: If it is only for kernel TCs,  let's remove unrelated CONFIG.

File: build/configs/esp32_DevKitC/kernel_tc/defconfig [hunk 2]
```diff
@@ -0,0 +1,839 @@
...
+# Application Configuration
+#
+
+#
+# Application entry point list
+#
+# CONFIG_ENTRY_MANUAL is not set
+CONFIG_ENTRY_HELLO=y
+# CONFIG_ENTRY_TESTCASE is not set
+CONFIG_USER_ENTRYPOINT="hello_main"
```
Comment (jeongchanKim) [inline build/configs/esp32_DevKitC/kernel_tc/defconfig]: This is tc config, but you set the entry point to hello_main. Is there any special reason?

File: build/configs/esp32_DevKitC/kernel_tc/defconfig [hunk 3]
```diff
@@ -0,0 +1,839 @@
...
+CONFIG_TC_KERNEL_LIBC_SIGNAL=y
+CONFIG_TC_KERNEL_LIBC_STDIO=y
+CONFIG_TC_KERNEL_LIBC_STDLIB=y
+CONFIG_TC_KERNEL_LIBC_STRING=y
+CONFIG_TC_KERNEL_LIBC_TIMER=y
+CONFIG_TC_KERNEL_LIBC_UNISTD=y
+CONFIG_TC_KERNEL_LIBC_SYSLOG=y
+CONFIG_TC_KERNEL_MQUEUE=y
+CONFIG_TC_KERNEL_PTHREAD=y
+# CONFIG_TC_KERNEL_ROUNDROBIN is not set
```
Comment (jeongchanKim) [inline build/configs/esp32_DevKitC/kernel_tc/defconfig]: Some TCs are disabled, such as roundrobin ,tash heapinfo and so on.

File: build/configs/esp_wrover_kit/kernel_tc/defconfig [hunk 4]
```diff
@@ -0,0 +1,839 @@
...
+# Application Configuration
+#
+
+#
+# Application entry point list
+#
+# CONFIG_ENTRY_MANUAL is not set
+CONFIG_ENTRY_HELLO=y
+# CONFIG_ENTRY_TESTCASE is not set
+CONFIG_USER_ENTRYPOINT="hello_main"
```
Comment (jeongchanKim) [inline build/configs/esp_wrover_kit/kernel_tc/defconfig]: Please check the entry point.

File: build/configs/esp32_DevKitC/kernel_tc/defconfig (hunk 1 above)
Comment (xiarihf) [reply build/configs/esp32_DevKitC/kernel_tc/defconfig]: It's created by make menuconfig, so I think it's better to keep all the items.

File: build/configs/esp32_DevKitC/kernel_tc/defconfig (hunk 2 above)
Comment (xiarihf) [reply build/configs/esp32_DevKitC/kernel_tc/defconfig]: Fixed. thanks.

File: build/configs/esp_wrover_kit/kernel_tc/defconfig (hunk 4 above)
Comment (xiarihf) [reply build/configs/esp_wrover_kit/kernel_tc/defconfig]: Fixed, thanks.

**Comment by seinfra** [comment]
Target : [3804a9945c962c0aa1a0508881cab1d59bc890df] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3804a9945c962c0aa1a0508881cab1d59bc890df] - Code Rule Check (C++) OK.

File: build/configs/esp32_DevKitC/kernel_tc/defconfig (hunk 3 above)
Comment (xiarihf) [reply build/configs/esp32_DevKitC/kernel_tc/defconfig]: @jeongchanKim Thanks for reminding us.

    Current defconfig file is created by "make menuconfig" operation. The HEAPINFO, STACKMONITOR, MEMORY_SAFETY, TERMIOS are closed in the default .config file which is created through menuconfig.
    Now, we enable the HEAPINFO, STACKMONITOR, MEMORY_SAFETY, but keep ROUND ROBIN 
 & TERMIOS  closed.  Becasue TERMIOS would be failed due to lack of some symbols in linking. And ROUND ROBIN is closed for we meeting crash issue in pthread stack overflow.
    We will check and fixed crash issue in round robin and the failed TC. 

Thanks.

File: build/configs/esp32_DevKitC/tc/Make.defs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+CXXFLAGS += -D_LIBCPP_HAS_NO_GENERALIZED_INITIALIZERS -fpermissive -mlongcalls
+
+
+
+CXXPICFLAGS = $(ARCHPICFLAGS) $(CXXFLAGS)
+CPPFLAGS = $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES)
+AFLAGS = $(CFLAGS) -D__ASSEMBLY__
+
+NXFLATLDFLAGS1 = -r -d -warn-common
+NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-pcrel.ld -no-check-sections
```
Comment (sunghan-chang) [inline build/configs/esp32_DevKitC/tc/Make.defs]: We don't support `NXFLAT`. Let's remove these lines.

File: build/configs/esp_wrover_kit/tc/Make.defs [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+
+
+CXXPICFLAGS = $(ARCHPICFLAGS) $(CXXFLAGS)
+CPPFLAGS = $(ARCHINCLUDES) $(ARCHDEFINES) $(EXTRADEFINES)
+AFLAGS = $(CFLAGS) -D__ASSEMBLY__
+
+NXFLATLDFLAGS1 = -r -d -warn-common
+NXFLATLDFLAGS2 = $(NXFLATLDFLAGS1) -T$(TOPDIR)/binfmt/libnxflat/gnu-nxflat-pcrel.ld -no-check-sections
+LDNXFLATFLAGS = -e main -s 2048
```
Comment (sunghan-chang) [inline build/configs/esp_wrover_kit/tc/Make.defs]: Same as previous

File: build/configs/esp32_DevKitC/tc/Make.defs (hunk 5 above)
Comment (sunghan-chang) [reply build/configs/esp32_DevKitC/tc/Make.defs]: I am sure. By another commit, I will remove them from makefile.export.

**Comment by seinfra** [comment]
Target : [d8b1c0027c885cb398e30b22ffc9da5549ba22b7] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d8b1c0027c885cb398e30b22ffc9da5549ba22b7] - Code Rule Check OK.

**Pull request merged**